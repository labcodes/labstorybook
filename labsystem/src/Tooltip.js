import React from "react";
import PropTypes from "prop-types";

export default class Tooltip extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired,
    placement: PropTypes.oneOf([
      "top-start",
      "top",
      "top-end",
      "right-start",
      "right",
      "right-end",
      "bottom-start",
      "left-start",
      "left",
      "left-end",
      "bottom",
      "bottom-end",
    ]).isRequired,
    color: PropTypes.string,
  };

  static defaultProps = {
    color: "black",
  };

  render() {
    const { id, text, color, children, placement } = this.props;

    return (
      <span className="lab-tooltip">
        {children}
        <span
          className={`lab-tooltip__text lab-tooltip--${color} lab-tooltip--${placement}`}
          id={id}
        >
          {text}
        </span>
      </span>
    );
  }
}
