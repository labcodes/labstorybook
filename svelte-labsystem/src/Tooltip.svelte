<script>
  export let id;
  export let text;
  export let placement = "top";

  $: if (text && text.length > 180) {
    console.warn(
      `Tooltip with id='${id}' has ${text.length} characters. It shouldn't be longer than 180 characters`
    );
  }
</script>

<span class="lab-tooltip">
  <slot />
  <span class={`lab-tooltip__text lab-tooltip--${placement}`} {id}>
    {text}
  </span>
</span>
