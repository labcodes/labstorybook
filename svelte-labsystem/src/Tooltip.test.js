/* eslint-disable no-console */
import html from "svelte-htm";
import { render } from "@testing-library/svelte";
import Tooltip from "./Tooltip.svelte";

describe("Tooltip", () => {
  const originalWarn = console.warn;
  afterEach(() => {
    console.warn = originalWarn;
  });

  it("renders with base props and placement top by default", async () => {
    const { container } = render(
      html`<${Tooltip} id="test" text="This is a Tooltip">child text<//>`
    );
    expect(container).toMatchSnapshot();
    expect(container.querySelectorAll(".lab-tooltip--top").length).toBe(1);
  });

  it("raises console.warn if text is longer than 180 chars", async () => {
    console.warn = jest.fn();
    render(
      html`<${Tooltip}
        id="test"
        text="Lorem ipsum dolor sit amet, has integre virtute consequat at. Sit quodsi gloriatur efficiantur in, quas inermis menandri ex vix. Sit quodsi gloriatur efficiantur in, quas inermis. Sit quodsi gloriatur."
        >child text<//
      >`
    );
    expect(console.warn).toBeCalled();
  });
});
