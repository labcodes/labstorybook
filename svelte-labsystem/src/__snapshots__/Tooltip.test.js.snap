// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders with base props and placement top by default 1`] = `
<body>
  <div>
    <span
      class="lab-tooltip"
    >
      child text
       
      <span
        class="lab-tooltip__text lab-tooltip--top"
        id="test"
      >
        This is a Tooltip
      </span>
    </span>
  </div>
</body>
`;
